cmake/partition_manager.cmake

```diff
@@ -428,10 +438,11 @@ else()
     )
 
   # Wrapper target for the merge command.
-  add_custom_target(merged_hex ALL DEPENDS ${final_merged})
-  # Add merged.hex as the representative hex file for flashing this app.
+  add_custom_target(merged_domains_hex ALL DEPENDS ${final_merged})
```

Comment:
How is this target used?

Reply:
It has the `ALL` argument, so its included in the default build. Without it the hex file would not be created since no other target depends on it.

Reply:
@oyvindronningstad Thanks for noticing this.

As @hakonfam points out, it's add to the default build, but what he might not be considering is that it is ONLY added to the default build, so if I do:
```
$ rm -rf build; cmake -GNinja -DBOARD=nrf5340pdk_nrf5340_cpuappns -Bbuild .
... A lot of CMake stuff
$  ninja -C build/ flash
.... A lot of build and flashing stuff .....
$  ll build/zephyr/*.hex
-rw-rw-r-- 1 tora tora 401K Jun 12 07:40 build/zephyr/merged.hex
-rw-rw-r-- 1 tora tora 320K Jun 12 07:40 build/zephyr/zephyr.hex
```
then there is NO merged_domains.hex, cause the default target was never built, but `merged.hex` was built, because it is required by `flash` target.
Not sure this was intended @hakonfam ?

If I instead do:
```
$  ninja -C build/ 
.... A lot of building .....
$  ll build/zephyr/*.hex
-rw-rw-r-- 1 tora tora 691K Jun 12 07:46 build/zephyr/merged_domains.hex
-rw-rw-r-- 1 tora tora 401K Jun 12 07:40 build/zephyr/merged.hex
-rw-rw-r-- 1 tora tora 320K Jun 12 07:40 build/zephyr/zephyr.hex
```
you will now see that `merged_domains.hex` is also created.

Secondly, I see that even when building for `nrf9160dk_nrf9160ns`, it creates a `merged_domains.hex` that is identical to `merged.hex`, I think the `merged_domains.hex` should only be created if there are multiple domains defined, as was proposed in #2461 
```